### ISSUE TITLE
Introduce a event for before the page starts rendering

### ISSUE THREAD

**Issue by omgitstom** [okta/okta-signin-widget#457]
There is a flash of content when detecting page changes:

```
    signIn.on('pageRendered', function (data) {
      if (data.page == "password-expired") {
          window.location = "https://google.com";
      }
    });
```

We need an event to be fired before the page starts rendering for people that want to override behavior before it loads.

**Comment by jmelberg-okta** [comment]
Closing as a duplicate of #539 - as we'll be tracking engineering work there.

**Comment by haishengwu-okta** [comment]
@omgitstom 

Do you have examples / background why people want to override behavior and what they want to do themselves instead of using signin widget default behavior?

**Comment by nbarbettini** [comment]
The `ready` event will provide this functionality, and is being released in [2.15](https://github.com/okta/okta-signin-widget/milestone/1).
